app/helpers/form_helper.rb

```diff
@@ -7,20 +9,46 @@ def app_setting_value_field_for(app_setting, f)
     end
   end
 
-  def block_text_field_tag(name, value)
-    text_field_tag(name, value, class: 'block col-12 mb2 field')
+  def block_text_field_tag(name, value, options = {})
+    text_field_tag(name, value, options.merge(class: 'block col-12 mb2 field')) +
+      form_input_error_messages(name, options)
   end
 
-  def block_date_field_tag(name, value)
-    date_field_tag(name, value, class: 'block col-12 mb2 field')
+  def block_date_field_tag(name, value, options = {})
+    date_field_tag(name, value, options.merge(class: 'block col-12 mb2 field')) +
+      form_input_error_messages(name, options)
   end
 
-  def us_states_territories_select_tag
+  # rubocop:disable MethodLength
+  def form_input_error_messages(name, options = {})
+    content_tag(:div, nil, class: 'bold red mb2', data: { 'errors-for' => name }) do
+      if options[:required]
+        concat content_tag(
+          :div,
+          t('forms.value_missing'),
+          style: 'display: none',
+          data: { 'errors-when' => 'valueMissing' }
+        )
+      end
+      if options[:pattern]
+        concat content_tag(
+          :div,
+          options[:'data-custom-message'] || t('forms.pattern_mismatch'),
+          style: 'display: none',
+          data: { 'errors-when' => 'patternMismatch' }
+        )
+      end
+    end
+  end
+  # rubocop:enable MethodLength
+
+  def us_states_territories_select_tag(options = {})
     select_tag(
       'state',
       options_for_select(us_states_territories),
-      class: 'block col-12 mb2 field'
-    )
+      options.merge(class: 'block col-12 mb2 field')
+    ) +
+      form_input_error_messages('state', options)
   end
```

Comment:
@monfresh that's a great idea for custom form inputs. This PR was just extending the poor direction (mine) of putting the class strings in the helper. Could we merge this as-is and then tackle the custom form input classes as a subsequent refactor? I'd be happy to tackle that.